### PR TITLE
	bugfix(fragment): minor fix for fragment warning

### LINKED ISSUE #8: UserProfile fragment compiled twice when provided as external
Providing custom userfield fragments causes Apollo to print a warning about a fragment that being compiled twice with the same name. 
Probably we need to compile it with `gql` tag only when it not provided at all

### PR DESCRIPTION
**Pull request by dotansimha**
Closes #8 

Also: added user field to refresh token request in order to return the valid user

### CODE REVIEW

File: packages/graphql-client/src/graphql-client.js [hunk 1]
```diff
@@ -12,20 +12,24 @@ import {
...
   defaultUserFieldsFragment,
+  createLoginMutation,
+  createRefreshTokenMutation,
 } from './graphql';
 
-
 export type OptionsType = {
   graphQLClient: any,
-  userFieldsFragment: string
+  userFieldsFragment: string,
```
Comment (davidyaha) [inline packages/graphql-client/src/graphql-client.js:22]: Does it lint? I think there was a rule against those

File: packages/graphql-client/src/graphql/login.mutation.js [hunk 2]
```diff
@@ -5,14 +5,13 @@ export const createLoginMutation = userFieldsFragment => gql`
   mutation($user: String!, $password: String!) {
     loginWithPassword(user: $user, password: $password) {
       sessionId
+      ...LoginFields
       user {
-        id
-        ...UserFields
+          ...UserFields
```
Comment (davidyaha) [inline packages/graphql-client/src/graphql/login.mutation.js]: why 4 space indents?

File: packages/graphql-client/src/graphql/login.mutation.js [hunk 3]
```diff
@@ -5,14 +5,13 @@ export const createLoginMutation = userFieldsFragment => gql`
   mutation($user: String!, $password: String!) {
     loginWithPassword(user: $user, password: $password) {
       sessionId
+      ...LoginFields
       user {
-        id
```
Comment (davidyaha) [inline packages/graphql-client/src/graphql/login.mutation.js]: I think we should try to ask for `id` regardless of what user fragment we get later

File: packages/graphql-client/src/graphql/user-fields.fragment.js [hunk 4]
```diff
@@ -1,6 +1,4 @@
-import gql from 'graphql-tag';
-
-export const defaultUserFieldsFragment = gql`
+export const defaultUserFieldsFragment = `
   fragment UserFields on User {
     email
```
Comment (davidyaha) [inline packages/graphql-client/src/graphql/user-fields.fragment.js:3]: Also the `id` field is not asked here

File: packages/graphql-client/src/graphql-client.js [hunk 5]
```diff
@@ -45,11 +49,19 @@ export class GraphQLClient {
       });
   }
 
+  async query(query, resultField, variables) {
```
Comment (davidyaha) [inline packages/graphql-client/src/graphql-client.js:52]: Didn't see any uses for `query`, for future use?

Comment (dotansimha) [reply packages/graphql-client/src/graphql-client.js:52]: Yeah :)

File: packages/graphql-client/src/graphql-client.js (hunk 1 above)
Comment (dotansimha) [reply packages/graphql-client/src/graphql-client.js:22]: Yeah it does :D

File: packages/graphql-client/src/graphql/login.mutation.js (hunk 3 above)
Comment (dotansimha) [reply packages/graphql-client/src/graphql/login.mutation.js]: Fixed

File: packages/graphql-client/src/graphql/login.mutation.js (hunk 2 above)
Comment (dotansimha) [reply packages/graphql-client/src/graphql/login.mutation.js]: Fixed

File: packages/graphql-client/src/graphql/user-fields.fragment.js (hunk 4 above)
Comment (dotansimha) [reply packages/graphql-client/src/graphql/user-fields.fragment.js:3]: Fixed

**Review by davidyaha — APPROVED** [review]

**Pull request merged**